frame/contracts/src/wasm/runtime.rs

```diff
@@ -2821,4 +2827,28 @@ pub mod env {
 		ctx.charge_gas(RuntimeCosts::InstantationNonce)?;
 		Ok(ctx.ext.nonce())
 	}
+
+	#[unstable]
+	fn add_delegate_dependency(
+		ctx: _,
+		memory: _,
+		code_hash_ptr: u32,
+	) -> Result<ReturnCode, TrapReason> {
+		ctx.charge_gas(RuntimeCosts::AddDelegateDependency)?;
+		let code_hash = ctx.read_sandbox_memory_as(memory, code_hash_ptr)?;
+		ctx.ext.add_delegate_dependency(code_hash)?;
+		Ok(ReturnCode::Success)
+	}
+
+	#[unstable]
+	fn remove_delegate_dependency(
+		ctx: _,
+		memory: _,
+		code_hash_ptr: u32,
+	) -> Result<ReturnCode, TrapReason> {
```

Comment:
Function is infallible (diverges on error).

```suggestion
	) -> Result<(), TrapReason> {
```